File: python/day2.py
with open('02/day2.txt', 'r') as f:
  lines = f.readlines()

entries = [tuple(string.strip().split(' ')) for string in lines]

horiz = 0
depth = 0
aim = 0
for entry in entries:
    action, value = entry
    if action == 'forward':
        horiz += int(value)
        depth += (aim * int(value))
    elif action == 'backward':
        horiz -= int(value)
        depth -= (aim * int(value))
    elif action == 'up':
      aim -= int(value)
    else:
      aim += int(value)

answer = horiz * depth
print(answer)

File: cpp/day2.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

#include "utils.hh"

using namespace std;

vector<string> split_commands(string line)
{
  return split_string(line, " ");
}

int main() {
  vector<string> lines = open_file("02/day2.txt");
  vector<vector<string> > entries;
  transform(lines.begin(), lines.end(), back_inserter(entries), split_commands);
  
  int horiz = 0;
  int depth = 0;
  int aim = 0;
  for (int i = 0; i < entries.size(); ++i) {
    string action = entries[i][0];
    int value = stoi(entries[i][1]);

    if (action == "forward") {
      horiz += value;
      depth += (aim * value);
    } else if (action == "backward") {
      horiz -= value;
      horiz -= (aim * value);
    } else if (action == "up") {
      aim -= value;
    } else {
      aim += value;
    }
  }

  int answer = { horiz * depth };
  cout << answer << endl;
  return answer;
}